src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -3090,6 +3267,57 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
         eval quick_visualization_preview((value) model.nodes.set_quick_preview(*value));
     }
 
+
+
+    // =============
+    // === Modes ===
+    // =============
+
+    let profiling_mode_transition = Animation::new(network);
+    frp::extend! { network
+        out.source.mode <+ frp.toggle_profiling_mode.map2(&frp.mode,|_,&mode| {
+            match mode {
+                Mode::Profiling => Mode::Normal,
+                _               => Mode::Profiling,
+            }
+        });
+
+        eval out.mode([model](_) {
+            for edge_id in model.edges.keys() {
+                model.refresh_edge_color(edge_id);
+            }
```

Comment:
As previously-  refattor it to a model function, like `refresh_all_edges_colors` - the FRP network def should be as short as possible so its easy to follow the FRP definition